.github/workflows/integration-unreleased.yml

```diff
@@ -36,17 +36,25 @@ jobs:
         working-directory: opensearch
         run: echo key=`git log -1 --format='%H'` >> $GITHUB_OUTPUT
 
+
       - name: Restore cached build
         id: cache-restore
         uses: actions/cache/restore@v3
         with:
           path: opensearch/distribution/archives/linux-tar/build/distributions
           key: ${{ steps.get-key.outputs.key }}
 
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        if: steps.cache-restore.outputs.cache-hit != 'true'
+        with:
+          distribution: 'corretto'
```

Comment:
We use `temurin` everwhere else. 

You need a default, so `${{ matrix.entry.jdk_version || 17 }}`.